core/message_pool.go

```diff
@@ -56,6 +56,13 @@ func (pool *MessagePool) Pending() []*types.SignedMessage {
 	return out
 }
 
+// Get retrieves a message from the pool by CID.
+func (pool *MessagePool) Get(c cid.Cid) *types.SignedMessage {
```

Comment:
I may be in the minority here but my opinion is that this interface would be more intuitive if it returned an error in the case that the pool does not contain the requested cid.

When called later from commands (or elsewhere) this error could then be propagated and returned to the user.

I understand the desire to use the zero value as a more concise indicator that the message isn't found, but for some reason it feels less golang-like.  One nice thing about returning an error when the message isn't found is that we can easily distinguish lack of membership from the should-never-happen bad-state situation where the message pool maps a legitimate cid to nil.

Interested to hear other opinions as I'm not sure if this is a mostly-good, mostly-bad, or neutral opinion from the standard golang style point of view.

Reply:
I would add that doing it this way breaks a convention in the code base, and so it violates the principle of least surprise.